src/engine/SpatialJoinAlgorithms.h

```diff
@@ -19,6 +19,7 @@
 
 #include "engine/Result.h"
 #include "engine/SpatialJoin.h"
+#include "global/RuntimeParameters.h"
```

Comment:
This header is very expensive. Consider moving it to the `.cpp` file (and the code along with it).